implementations/rust/ockam/ockam_api/src/cli_state.rs

```diff
@@ -258,6 +258,34 @@ impl IdentitiesState {
         Ok(IdentityState { path, config })
     }
 
+    pub fn list(&self) -> Result<Vec<IdentityState>> {
+        let mut identities = vec![];
+        for entry in std::fs::read_dir(&self.dir)? {
+            let entry = entry?;
+            let filename = entry.file_name().into_string()
+                .map_err(|_| CliStateError::Io(std::io::Error::new(
+                std::io::ErrorKind::InvalidData,
+                "identity has an invalid name",
+            )))?;
+            if !entry.file_type()?.is_dir() && filename.ends_with(".json") {
+                let identity_name = &filename[0..filename.len()-5];
+                identities.push(self.get(identity_name)?);
+            }
+        }
+        Ok(identities)
+    }
+
+    pub fn delete(&self, name: &str) -> Result<()> {
+        let identity_state = self.get(name)?;
+        std::fs::remove_file(identity_state.path)?;
+        let identifier = identity_state.config.identifier.to_string();
+        let default = self.default()?;
+        if default.config.identifier.to_string() == identifier {
+            let _ = std::fs::remove_file(self.default_path()?);
+        }
+        Ok(())
```

Comment:
This could also be simplified. The `IdentityIdentifier` implements the `PartialEq` trait, so you can compare them without having to convert them into strings. Also, you can remove some redundant variables (like `default`):

```rust
let identity = self.get(name)?;
std::fs::remove_file(identity.path)?;
if self.default()?.config.identifier == identity.config.identifier {
    let _ = std::fs::remove_file(self.default_path()?);
}
```